samples/bluetooth/direct_test_mode/src/transport/hci_uart.c

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/net/buf.h>
+#include <zephyr/bluetooth/hci_types.h>
+#include <zephyr/sys/byteorder.h>
+
+#include "dtm_hci.h"
+#include "hci_uart.h"
+
+#ifndef CONFIG_DTM_TRANSPORT_LOG_LEVEL
+#define CONFIG_DTM_TRANSPORT_LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#endif
+
+#ifdef CONFIG_DTM_TRANSPORT_HCI
+#define QUEUE_COUNT CONFIG_DTM_HCI_QUEUE_COUNT
+#define QUEUE_SIZE CONFIG_DTM_HCI_QUEUE_SIZE
+#define TX_THREAD_STACK_SIZE CONFIG_DTM_HCI_TX_THREAD_STACK_SIZE
+#define TX_THREAD_PRIORITY CONFIG_DTM_HCI_TX_THREAD_PRIORITY
+#else
+#define QUEUE_COUNT CONFIG_REMOTE_HCI_QUEUE_COUNT
+#define QUEUE_SIZE CONFIG_REMOTE_HCI_QUEUE_SIZE
+#define TX_THREAD_STACK_SIZE CONFIG_REMOTE_HCI_TX_THREAD_STACK_SIZE
+#define TX_THREAD_PRIORITY CONFIG_REMOTE_HCI_TX_THREAD_PRIORITY
+#endif
+
+LOG_MODULE_REGISTER(dtm_hci_uart, CONFIG_DTM_TRANSPORT_LOG_LEVEL);
+
+#define DTM_UART DT_CHOSEN(ncs_dtm_uart)
+static const struct device *hci_uart_dev = DEVICE_DT_GET(DTM_UART);
+
+NET_BUF_POOL_DEFINE(hci_tx_buf, QUEUE_COUNT, QUEUE_SIZE, 0, NULL);
+static K_FIFO_DEFINE(hci_tx_queue);
+
+NET_BUF_POOL_DEFINE(hci_rx_buf, QUEUE_COUNT, QUEUE_SIZE, sizeof(uint8_t), NULL);
+
+enum h4_state {
+	S_TYPE,
+	S_HEADER,
+	S_PAYLOAD
+};
+
+static hci_uart_read_cb dtm_hci_put;
+
+static int hci_hdr_len(uint8_t type)
+{
+	switch (type) {
+	case H4_TYPE_CMD:
+		return sizeof(struct bt_hci_cmd_hdr);
+
+	case H4_TYPE_ACL:
+		return sizeof(struct bt_hci_acl_hdr);
+
+	case H4_TYPE_EVT:
+		return sizeof(struct bt_hci_evt_hdr);
+
+	case H4_TYPE_ISO:
+		return sizeof(struct bt_hci_iso_hdr);
+
+	default:
+		return 0;
+	}
+}
+
+static int hci_pld_len(uint8_t type, uint8_t *hdr)
+{
+	switch (type) {
+	case H4_TYPE_CMD:
+		return ((struct bt_hci_cmd_hdr *)hdr)->param_len;
+
+	case H4_TYPE_ACL:
```

Comment:
Do you need ACL and ISO support?

Reply:
Support for packet is not needed, but support for receiving and parsing is needed to correctly find the beginings and ends of other packets.